### PR TITLE
Dev 1545 env vars

### PR DESCRIPTION
**Pull request by njharman**
#### What's this PR do?
Reorganize settings into categories; django, apps, 1st party, etc
Document settings, apps.
Standardize on os.getenv over os.environ.get
Move all environment access to settings.

#### Why are we doing this? How does it help us?
Documentation.  Onboarding new devs, even old devs trying to understand code/debug/etc. 
Clearer to have all the "external" environment access in one place.

#### How should this be manually tested?
N/A

#### Have automated unit tests been added?
N/A

#### How should this change be communicated to end users?
N/A

#### Are there any smells or added technical debt to note?
N/A

#### Has this been documented? If so, where?
It is documentation!

#### What are the relevant tickets? Add a link to any relevant ones.
[DEV-1545]

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
no, should not be. But since lots of environment variable code touched, there is a higher chance of regression/introduced bugs.

#### Are there next steps? If so, what? Have tickets been opened for them?
no, but going forward we should add settings to appropriate sections and add comments linking settings to their apps/purpose.

[DEV-1545]: https://news-revenue-hub.atlassian.net/browse/DEV-1545?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: revengine/settings/base.py [hunk 1]
```diff
@@ -10,22 +9,75 @@
...
+USE_DEBUG_INTERVALS = os.getenv("USE_DEBUG_INTERVALS", False)
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 PROJECT_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 BASE_DIR = os.path.dirname(PROJECT_DIR)
 
-# Quick-start development settings - unsuitable for production
-# See https://docs.djangoproject.com/en/3.0/howto/deployment/checklist/
+# https://news-revenue-hub.atlassian.net/wiki/spaces/TECH/pages/2014871553/Rev+Engine+NRE+development+deployment+servers+and+environments
+ENVIRONMENT = os.getenv("ENVIRONMENT", "dev")
```
Comment (x110dc) [inline revengine/settings/base.py:26]: This should be set to `unknown` so we can distinguish Sentry alerts from unconfigured systems from DEV. I encourage devs to set `ENVIRONMENT` locally to their names.

File: revengine/settings/base.py [hunk 2]
```diff
@@ -10,22 +9,75 @@
...
+
+# Internationalization
+# https://docs.djangoproject.com/en/3.0/topics/i18n/
+LANGUAGE_CODE = "en-us"
+TIME_ZONE = "UTC"
+USE_I18N = True
+USE_L10N = True
+USE_TZ = True
+# COUNTRIES is not Djanog setting, is used in model choice fields, related to I18N.
+# First in this list will be default.
```
Comment (x110dc) [inline revengine/settings/base.py:52]: "Djanog" typo.

File: revengine/settings/base.py [hunk 3]
```diff
@@ -10,22 +9,75 @@
...
+# Use the 2-char country code here for Stripe's sake.
+COUNTRIES = ["US", "CA"]  # First in this list will be default.
+# CURRENCIES is not Djanog setting, is used in models, related to I18N.
```
Comment (x110dc) [inline revengine/settings/base.py:55]: "Djanog" typo again.

File: revengine/settings/base.py [hunk 4]
```diff
@@ -10,22 +9,75 @@
...
+# Map currency-code to symbol.
+CURRENCIES = {"USD": "$", "CAD": "$"}
+
+# Static files (CSS, JavaScript, Images)
+# https://docs.djangoproject.com/en/3.0/howto/static-files/
+STATIC_ROOT = os.path.join(BASE_DIR, "public/static")
```
Comment (x110dc) [inline revengine/settings/base.py:61]: Should we make the link to 3.2 instead of 3.0? Not that important.

File: revengine/settings/base.py [hunk 5]
```diff
@@ -45,24 +97,23 @@
...
     "django_filters",
+    "django_json_widget",
+    "rest_framework",
     "sorl.thumbnail",
     "sorl_thumbnail_serializer",
-    "solo",
-    "anymail",
-    "django_json_widget",
-    "health_check",
+    "solo",  # Single row models, e.g. HubSlackInteration.
```
Comment (x110dc) [inline revengine/settings/base.py:105]: Typo "HubSlackInteration"

File: revengine/settings/base.py [hunk 6]
```diff
@@ -45,24 +97,23 @@
...
+    "anymail",  # Email service provider integration.
+    "health_check",  # Checks for various conditions and provides reports when anomalous behavior is detected.
     "health_check.db",
     "health_check.cache",
     "health_check.contrib.migrations",
     "health_check.contrib.redis",
-    "waffle",
-    "reversion",
+    "waffle",  # Djanog feature flag support.
```
Comment (x110dc) [inline revengine/settings/base.py:112]: "Djanog" typo.

File: revengine/settings/base.py [hunk 7]
```diff
@@ -405,29 +314,135 @@
...
+    "ACCESS_TOKEN_LIFETIME": timedelta(hours=float(os.getenv("ACCESS_TOKEN_LIFETIME_HOURS", 12))),
+    "REFRESH_TOKEN_LIFETIME": timedelta(days=2),
+}
+
+SECURE_SSL_REDIRECT = os.getenv("SECURE_SSL_REDIRECT", "true").lower() == "true"
+
+### sorl-thumbmail Settings
+THUMBNAIL_COLORSPACE = None
+THUMBNAIL_PRESERVE_FORMAT = True
+THUMBNAIL_PRESERVE_FORMAT = True
```
Comment (x110dc) [inline revengine/settings/base.py:329]: Duplicate line.

File: revengine/settings/base.py [hunk 8]
```diff
@@ -405,29 +314,135 @@
...
+
+
+### reversion Settings
+# Add reversion models to admin interface.
+ADD_REVERSION_ADMIN = True
+
 
-SECURE_SSL_REDIRECT = os.environ.get("SECURE_SSL_REDIRECT", "true").lower() == "true"
+### Stripe Settings
+STRIPE_API_VERSION = "2020-08-27"  # Stripe API Target Version
```
Comment (x110dc) [inline revengine/settings/base.py:338]: Could add a link here to https://stripe.com/docs/upgrades#api-changelog

File: revengine/settings/base.py [hunk 9]
```diff
@@ -405,29 +314,135 @@
...
+# for the `gtm_id` context key in `revengine.views.ReactAppView`. In turn, spa.public.index.html
+# references `gtm_id`. If this value is defined, a Google Tag Manager script is added to head and GTM
+# iframe added to body.
 HUB_GTM_ID = os.getenv("HUB_GTM_ID")
 
 
-# Sentry Configuration
-SENTRY_ENABLE_FRONTEND = os.environ.get("SENTRY_ENABLE_FRONTEND", "false").lower() == "true"
-SENTRY_DSN_FRONTEND = os.environ.get("SENTRY_DSN_FRONTEND")
+### Heroku Settings
```
Comment (x110dc) [inline revengine/settings/base.py:371]: These are only used in the bootstrapping script for review apps. Should probably note that here.

File: revengine/settings/base.py [hunk 10]
```diff
@@ -405,29 +314,135 @@
...
+HEROKU_APP_NAME = os.getenv("HEROKU_APP_NAME")
+HEROKU_BRANCH = os.getenv("HEROKU_BRANCH")
+CF_ZONE_NAME = os.getenv("CF_ZONE_NAME")
+
 
-SENTRY_ENABLE_BACKEND = os.environ.get("SENTRY_ENABLE_BACKEND", "false").lower() == "true"
-SENTRY_DSN_BACKEND = os.environ.get("SENTRY_DSN_BACKEND")
+### Django-phonenumber Settings
+# TODO: These appear to be for django-phonenumber-field which isn't in apps, isn't imported, but is in poetry.
```
Comment (x110dc) [inline revengine/settings/base.py:378]: Can we add tickets for these TODOs?

File: revengine/settings/base.py [hunk 11]
```diff
@@ -405,29 +314,135 @@
...
+METADATA_SOURCE = os.getenv("METADATA_SOURCE", "rev-engine")
+METADATA_SCHEMA_VERSION = os.getenv("METADATA_SCHEMA_VERSION", "1.0")
+
+
+# This is the interval at which flagged payments will be automatically captured.
+# NOTE: Stripe automatically REJECTS flagged payments every 7 days. Make sure
+# this delta is less than 6.5 days to be safe.
+FLAGGED_PAYMENT_AUTO_ACCEPT_DELTA = timedelta(days=3)
+
+# TODO: USER_TTL is not referenced anywhere in codebase.
```
Comment (x110dc) [inline revengine/settings/base.py:404]: Can we create a ticket for this TODO?

File: revengine/settings/base.py [hunk 12]
```diff
@@ -405,29 +314,135 @@
...
+DEFAULT_FROM_EMAIL = f"noreply@{os.getenv('DOMAIN', 'example.com')}"
+EMAIL_BACKEND = "django.core.mail.backends.console.EmailBackend"
+EMAIL_SUBJECT_PREFIX = "[RevEngine %s] " % ENVIRONMENT.title()
+# Revengine template identifiers
+EMAIL_DEFAULT_TRANSACTIONAL_SENDER = os.getenv(
+    "EMAIL_DEFAULT_TRANSACTIONAL_SENDER", "News Revenue Engine <no-reply@fundjournalism.org>"
+)
+
+## BadActor API
+BAD_ACTOR_API_URL = os.getenv("BAD_ACTOR_API_URL", "https://bad-actor-test.fundjournalism.org/v1/bad_actor/")
```
Comment (x110dc) [inline revengine/settings/base.py:439]: I don't think we should have this default in the code.

File: revengine/settings/base.py [hunk 13]
```diff
@@ -405,29 +314,135 @@
...
+# NOTE: We've been given keys with some characters that might need escaping as environment variables, eg "$"
+BAD_ACTOR_API_KEY = os.getenv("BAD_ACTOR_API_KEY", "testing_123")
```
Comment (x110dc) [inline revengine/settings/base.py:441]: Is there a reason to have a default here?

**Review by x110dc — CHANGES_REQUESTED** [review]

File: revengine/settings/base.py (hunk 1 above)
Comment (njharman) [reply revengine/settings/base.py:26]: Done

File: revengine/settings/base.py (hunk 4 above)
Comment (njharman) [reply revengine/settings/base.py:61]: yes

File: revengine/settings/base.py (hunk 11 above)
Comment (njharman) [reply revengine/settings/base.py:404]: cutting it instead. also not mentioned anywhere in ./site-packages/...

File: revengine/settings/base.py (hunk 12 above)
Comment (njharman) [reply revengine/settings/base.py:439]: replaced with empty default

File: revengine/settings/base.py (hunk 13 above)
Comment (njharman) [reply revengine/settings/base.py:441]: replaced with empty default

File: revengine/settings/base.py (hunk 9 above)
Comment (njharman) [reply revengine/settings/base.py:371]: That's not true.

HEROKU_APP_NAME in common/utils.py test ofr same. HEROKU_BRANCH in teardown-review-app.py. CF_ZONE_NAME in all of those.

File: revengine/settings/base.py (hunk 10 above)
Comment (njharman) [reply revengine/settings/base.py:378]: To what? Investigate if django-phonenumber is used or not?

Rather just delete it now, Or replace todo with explanation if you know it is needed.

File: revengine/settings/base.py (hunk 12 above)
Comment (njharman) [reply revengine/settings/base.py:439]: Actually removing either default caused test failures. Removing them and fixing tests should be separate ticket.

File: revengine/settings/base.py (hunk 13 above)
Comment (njharman) [reply revengine/settings/base.py:441]: Actually removing either default caused test failures. Removing them and fixing tests should be separate ticket.

File: revengine/settings/base.py (hunk 10 above)
Comment (x110dc) [reply revengine/settings/base.py:378]: AFAIK it's not used. Let's delete.

File: revengine/settings/base.py (hunk 12 above)
Comment (x110dc) [reply revengine/settings/base.py:437]: I'll open a ticket for it.

**Review by x110dc — CHANGES_REQUESTED** [review]

**Pull request closed without merging**